pageserver/src/tenant/timeline/walreceiver/walreceiver_connection.rs

```diff
@@ -434,9 +434,7 @@ fn ignore_expected_errors(pg_error: postgres::Error) -> anyhow::Result<postgres:
     {
         return Ok(pg_error);
     } else if let Some(db_error) = pg_error.as_db_error() {
-        if db_error.code() == &SqlState::CONNECTION_FAILURE
```

Comment:
Why `db_error.code()` is not checked anymore?

Reply:
Previously we sent io error in this case:
```
                            return Err(QueryError::from(io::Error::new(
                                io::ErrorKind::ConnectionAborted,
                                format!("end streaming to {:?}", spg.appname),
                            )));
```

That's not really an io error, so I changed to throw anyhow instead, which gets transformed into PG kisses and hugs `XXOOO` internal server error, not `CONNECTION_FAILURE`. That's perhaps not the most appropriate errcode as well, but better typing here can be made separately, it needs changed in postgres_backend API.

Reply:
> If someone will write an error message
That's extremely unlikely, but I've found there is a SUCCESSFUL_COMPLETION just for this case.

Reply:
> That's extremely unlikely

The words are not unusual, so it is unlikely, but can happen. 

Thanks for bringing back the check